### ISSUE TITLE
Replace `__Dymola_InlineNoEvent` to the new standard equivalent

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#748]
**Reported by dietmarw on 24 May 2012 15:23 UTC**
Since Modelica 3.3 there is a new standard to handle the old tool dependent annotation `__Dymola_InlineNoEvent`. So use the standard instead.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/748

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 27 Jul 2012 11:54 UTC**
Do I understand the new 3.3 annotation correctly when I assume that this is the correct replacement:
```diff
- annotation(__Dymola_InlineNoEvent=true)
+ annotation(Inline=true, GenerateEvent=false)
```

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 20 Aug 2012 08:05 UTC**
Almost. 

It is GenerateEvents not GenerateEvent; and "Inline=true" should not be needed (although there is nothing wrong with having it).

**Comment by modelica-trac-importer** [comment]
**Changelog removed by hansolsson on 20 Aug 2012 08:05 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 20 Aug 2012 08:19 UTC**
Sorry my mistake about the misspelling of `GenerateEvents`. So what you are saying is that the tool annotation `__Dymola_InlineNoEvent=true` does not implicitly "say" `Inline=true`. Have I understood that correctly? I just want to make sure that by using the standard annotation we don't end up with different simulation behaviour.

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 20 Aug 2012 08:28 UTC**
It is more that GenerateEvents also imply "Inline=true" (the spec states "one possibility of doing this is to inline the function and generate events for the inlined function" - and it is certainly the easiest way to implement it in this case).

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 20 Aug 2012 11:15 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 10 Sep 2012 15:06 UTC**
The changes applied in order to fix these tickets were reverted because we the MSL 3.2.1 finalisation was decided to happen in trunk. These MLS 3.3 fixes should be reaplied once `trunk` is used for MSL3.3 development again.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 3 Dec 2012 16:13 UTC**
Rescheduling since it now was decided to backport the Modelica 3.3 annotation to 3.2rev2.
The decision though was to keep the  `__Dymola_InlineNoEvent=false` and add the new `GenerateEvents=true` and then for the next MSL release simply remove `__Dymola_InlineNoEvent=false` finally.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 3 Dec 2012 17:22 UTC**
Fixed in 6fabc6be001000b0d2a9b6ae2675e245acfa5024  by adding `Inline=true,GenerateEvents=true`.

Except there still exist two classes which have the combination of `__Dymola_InlineNoEvent=false` and `Inline = false` which doesn't seem useful and Hans will look at it :)

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 3 Dec 2012 17:29 UTC**
Replying to [comment:9 dietmarw]:
> Fixed in 6fabc6be001000b0d2a9b6ae2675e245acfa5024  by adding `Inline=true,GenerateEvents=true`.
> 
> Except there still exist two classes which have the combination of `__Dymola_InlineNoEvent=false` and `Inline = false` which doesn't seem useful and Hans will look at it :)

If Inline=false then InlineNoEvent has no effect and it could just be removed.
I don't know why these functions have Inline=false.

Proposal: just remove InlineNoEvents=false if Inline=false.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 3 Dec 2012 18:59 UTC**
OK proposal implemented in 23ad2c5aef06d009400d7eaf90e527bbd7751dbe.
